obspy/io/mseed/tests/test_mseed_reading_and_writing.py

```diff
@@ -167,12 +168,25 @@ def test_downcast_int64_to_int32(self):
         st.append(tr)
 
         # make sure the data can be written to mseed -> write to io.BytesIO() object in test 
-        file = io.BytesIO()
-        st.write(file, format="mseed")
-        st2 = read(file)
+        with io.BytesIO() as buf:
+            st.write(buf, format="mseed")
+            st2 = read(buf)
+
+            res = np.array_equal(st,st2)
+            self.assertTrue(res)
+        
+        # Test that error message is indeed raised when data cannot be downcast
+        # Create dummy stream that cannot be properly downcast to int64
+        x_int64 = np.arange(2147483648, 9223372036854775807, 460000000000000000, dtype=np.int64)
```

Comment:
Ooof that's pretty long numbers to look at. I am assuming this arange will only contain a handful of numbers (it should only use a few so tests stay slim and fast)? But nobody will be able to tell from looking at this, so you should rather hard code it with like 3-4 numbers in it..

```
x_int64 = np.array([2147483648, ..., 9223372036854775807], dtype=np.int64)
```

Also, a good test should cover as many apects / potential inputs as possible. In this case, you should definitely add a test that has large negative numbers, as buggy code might not take that into account properly (I know yours does, but still -- tests should be agnostic of the code). Also, it seems the test data contains only numbers that can't be int32-represented. I think a better test data set would be to have like 4-5 small numbers and then just one large (incompatible) number among them.

Also, come to think of it, the check if data can besafely downcast to int32 could probably be replaced by checking the min/max values of the data (and checking against `np.iinfo(np.int32).min/max`). That might be faster, but we have it super safe I guess, so we'll leave as it for now I think.

Reply:
Good point, I've updated the test for negative numbers and a mix of mainly small numbers with one too big one. Also a good point that checking that the min and max are in the right range should be just as good, I suppose it would take about the same space, but maybe it's faster to compute?

Reply:
These seem like they'd be a bit clearer written as powers of 2/10: `2147483648, 9223372036854775807, 460000000000000000` == `2**31, 2**63 + 1, 460 * 10**15` (though maybe if the exact numbers don't matter too much, the last can be another power of 2 as well.)

Reply:
> Also a good point that checking that the min and max are in the right range should be just as good, I suppose it would take about the same space, but maybe it's faster to compute?

Just leave as is for now.. it's safe as it's implemented as is right now and that's most important for such an edge case.

Reply:
> the exact numbers don't matter too much, the last can be another power of 2 as well

Yep maybe clearer to just use powers of 2. And yes the exact numbers don't matter at all.